flows/routers/base.go

```diff
@@ -49,6 +50,11 @@ func (r *BaseRouter) Type() string { return r.type_ }
 // Wait returns the optional wait on this router
 func (r *BaseRouter) Wait() flows.Wait { return r.wait }
 
+// AllowTimeout returns whether this router can be resumed at with a timeout
+func (r *BaseRouter) AllowTimeout() bool {
+	return r.wait != nil && !utils.IsNil(r.wait.Timeout())
```

Comment:
punting on solving the interface to a nil problem for now with our usual solution